.pre-commit-config.yaml

```diff
@@ -10,12 +10,13 @@ repos:
     hooks:
     -   id: ruff
         types_or: [
-            python, 
-            pyi, 
+            python,
+            pyi,
             # jupyter, # https://github.com/xdslproject/xdsl/issues/1273
             ]
         args: [ --fix, --exit-non-zero-on-fix ]
--   repo: https://github.com/psf/black
+-   repo: https://github.com/psf/black-pre-commit-mirror
     rev: 23.3.0
     hooks:
     -   id: black
+        exclude: "/(__pycache__|.asv|venv|build|tests/filecheck)/|versioneer.py$|xdsl/_version.py$"
```

Comment:
I could not find a way to get the pre-commit hook to find the pyproject.toml, I'd love for someone to figure this out but I'm tapped out